build/index.js

```diff
@@ -266,12 +267,40 @@ function logSuccess(command, timeLapsed) {
     logMain(message);
 }
 
+function checkVersions(metaManifest) {
+    const thisNPM = process.platform === "win32" ? "npm.cmd" : "npm";
+    let exitCode = 0;
+    const nodeMajorVersion = parseInt(process.version.slice(1, 3), 10);
+    if (nodeMajorVersion < metaManifest.requiredNodeMajorVersion) {
+        console.error(
+            `⚠️ Node.js v${metaManifest.requiredNodeMajorVersion} or later ` +
+                `required, you are on ${process.version}`
+        );
+        exitCode = 1;
+    }
+    const npmVersion = execSync(`${thisNPM} --version`, { encoding: 'utf-8' });
+    const npmMajorVersion = npmVersion.match(/^(\d+)\.\d+/)[0];
+    if (npmMajorVersion < metaManifest.requiredNPMMajorVersion) {
+            console.error(
+                `⚠️ NPM ${metaManifest.requiredNPMMajorVersion} or later ` +
+                `required, you are on ${npmMajorVersion}`
+            );
+            exitCode = 1;
+    }
+    return exitCode;
+}
+
 async function main() {
     const { scriptPath, command, start } = parseCommandLine();
     if (!command) {
         showUsage(scriptPath);
         return 1;
     }
+    const metaManifest = parseManifest("build/packages.json");
```

Comment:
This code makes more sense up here, but I think it's a bit _too_ high: It should be below the `process.chdir` line below, otherwise `build.js` cannot be invoked from an arbitrary directory, because the meta manifest wouldn't be found.